example/src/main/java/org/nervos/ckb/ACPTransactionExample.java

```diff
@@ -0,0 +1,187 @@
+    Thread.sleep(30000);
+    System.out.println("Transfer SUDT tx hash: " + transfer(new OutPoint(acpHash, "0x0")));
+  }
+
+  private static BigInteger getBalance(String address) throws IOException {
+    return new IndexerCollector(api, ckbIndexerApi).getCapacity(address);
+  }
+
+  private static String createACPCell() throws IOException {
+    List<ScriptGroupWithPrivateKeys> scriptGroupWithPrivateKeysList = new ArrayList<>();
+
+    TransactionBuilder txBuilder = new TransactionBuilder(api);
+    IndexerCollector txUtils = new IndexerCollector(api, ckbIndexerApi);
+
+    List<Receiver> receivers =
+        Collections.singletonList(new Receiver(receiverAcpAddress, Utils.ckbToShannon(200)));
+    List<CellOutput> cellOutputs = txUtils.generateOutputs(receivers, SendAddresses.get(0));
+    txBuilder.addOutputs(cellOutputs);
+
+    txBuilder.setOutputsData(Arrays.asList("0x", "0x"));
+    txBuilder.addCellDep(new CellDep(new OutPoint(ACP_TX_HASH, "0x0"), CellDep.DEP_GROUP));
+
+    // You can get fee rate by rpc or set a simple number
+    BigInteger feeRate = BigInteger.valueOf(1024);
+
+    // initial_length = 2 * secp256k1_signature_byte.length
+    // collectInputsWithIndexer method uses indexer rpc to collect cells quickly
+    CollectResult collectResult =
+        txUtils.collectInputs(SendAddresses, txBuilder.buildTx(), feeRate, Sign.SIGN_LENGTH * 2);
+
+    // update change cell output capacity after collecting cells if there is changeOutput
+    if (Numeric.toBigInt(collectResult.changeCapacity).compareTo(MIN_CKB) >= 0) {
+      cellOutputs.get(cellOutputs.size() - 1).capacity = collectResult.changeCapacity;
+      txBuilder.setOutputs(cellOutputs);
+    }
+
+    int startIndex = 0;
+    for (CellsWithAddress cellsWithAddress : collectResult.cellsWithAddresses) {
+      txBuilder.addInputs(cellsWithAddress.inputs);
+      for (int i = 0; i < cellsWithAddress.inputs.size(); i++) {
+        txBuilder.addWitness(i == 0 ? new Witness(Witness.SIGNATURE_PLACEHOLDER) : "0x");
+      }
+      if (cellsWithAddress.inputs.size() > 0) {
+        scriptGroupWithPrivateKeysList.add(
+            new ScriptGroupWithPrivateKeys(
+                new ScriptGroup(
+                    NumberUtils.regionToList(startIndex, cellsWithAddress.inputs.size())),
+                Collections.singletonList(
+                    SendPrivateKeys.get(SendAddresses.indexOf(cellsWithAddress.address)))));
+        startIndex += cellsWithAddress.inputs.size();
+      }
+    }
+
+    Secp256k1SighashAllBuilder signBuilder = new Secp256k1SighashAllBuilder(txBuilder.buildTx());
+
+    for (ScriptGroupWithPrivateKeys scriptGroupWithPrivateKeys : scriptGroupWithPrivateKeysList) {
+      signBuilder.sign(
+          scriptGroupWithPrivateKeys.scriptGroup, scriptGroupWithPrivateKeys.privateKeys.get(0));
+    }
+    Transaction tx = signBuilder.buildTx();
+    return api.sendTransaction(tx);
+  }
+
+  private static String transfer(OutPoint acpOutPoint) throws IOException {
```

Comment:
If the `transfer` function accepts an `outpoint`, We may need to add an example of collecting the ACP outpoint.

Reply:
The `acpOutPoint` comes from the result of `createAcpCell` function.

Reply:
This is to reduce complexity

Reply:
got it.